test_d_retry problem on yellowstone

There seems to be a problem with this test on yellowstone, the TESTBUILDFAIL_P1.f19_g16_rx1.A.yellowstone_intel
PASSES on the second attempt, I can reproduce this behavior consistantly using scripts_regression_tests.py but If I just run TESTBUILDFAIL_P1.f19_g16_rx1.A.yellowstone_intel it
fails everytime (as expected).   Any ideas?

```
======================================================================
FAIL: test_d_retry (__main__.O_TestTestScheduler)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./scripts_regression_tests.py", line 1070, in test_d_retry
    self._create_test(args)
  File "./scripts_regression_tests.py", line 813, in _create_test
    expected_stat=expected_stat)
  File "./scripts_regression_tests.py", line 60, in run_cmd_assert_result
    test_obj.assertEqual(stat, expected_stat, msg=msg)
AssertionError: 
COMMAND:  /glade/p/work/jedwards/sandboxes/cesm2_0_alpha/cime/scripts/create_test TESTBUILDFAIL_P1.f19_g16_rx1.A TESTRUNFAIL_P1.f19_g16_rx1.A TESTRUNPASS_P1.f19_g16_rx1.A --retry=1 -t 20171109_140326 --baseline-root /glade/scratch/jedwards/scripts_regression_test.20171109_140320/baselines --test-root=/glade/scratch/jedwards/scripts_regression_test.20171109_140320 --output-root=/glade/scratch/jedwards/scripts_regression_test.20171109_140320
FROM_DIR: /glade/p/work/jedwards/sandboxes/cesm2_0_alpha/cime/scripts/tests
SHOULD HAVE WORKED, INSTEAD GOT STAT 100
OUTPUT: Testnames: ['TESTBUILDFAIL_P1.f19_g16_rx1.A.yellowstone_intel', 'TESTRUNFAIL_P1.f19_g16_rx1.A.yellowstone_intel', 'TESTRUNPASS_P1.f19_g16_rx1.A.yellowstone_intel']
Using project from env PROJECT: CESM0008
Creating test directory /glade/scratch/jedwards/scripts_regression_test.20171109_140320/TESTBUILDFAIL_P1.f19_g16_rx1.A.yellowstone_intel.20171109_140326
Creating test directory /glade/scratch/jedwards/scripts_regression_test.20171109_140320/TESTRUNFAIL_P1.f19_g16_rx1.A.yellowstone_intel.20171109_140326
Creating test directory /glade/scratch/jedwards/scripts_regression_test.20171109_140320/TESTRUNPASS_P1.f19_g16_rx1.A.yellowstone_intel.20171109_140326
RUNNING TESTS:
  TESTBUILDFAIL_P1.f19_g16_rx1.A.yellowstone_intel
  TESTRUNFAIL_P1.f19_g16_rx1.A.yellowstone_intel
  TESTRUNPASS_P1.f19_g16_rx1.A.yellowstone_intel
Starting CREATE_NEWCASE for test TESTBUILDFAIL_P1.f19_g16_rx1.A.yellowstone_intel with 1 procs
Starting CREATE_NEWCASE for test TESTRUNFAIL_P1.f19_g16_rx1.A.yellowstone_intel with 1 procs
Starting CREATE_NEWCASE for test TESTRUNPASS_P1.f19_g16_rx1.A.yellowstone_intel with 1 procs
Finished CREATE_NEWCASE for test TESTBUILDFAIL_P1.f19_g16_rx1.A.yellowstone_intel in 2.535957 seconds (PASS)
Finished CREATE_NEWCASE for test TESTRUNPASS_P1.f19_g16_rx1.A.yellowstone_intel in 2.551518 seconds (PASS)
Finished CREATE_NEWCASE for test TESTRUNFAIL_P1.f19_g16_rx1.A.yellowstone_intel in 2.552595 seconds (PASS)
Starting XML for test TESTBUILDFAIL_P1.f19_g16_rx1.A.yellowstone_intel with 1 procs
Starting XML for test TESTRUNFAIL_P1.f19_g16_rx1.A.yellowstone_intel with 1 procs
Starting XML for test TESTRUNPASS_P1.f19_g16_rx1.A.yellowstone_intel with 1 procs
Finished XML for test TESTBUILDFAIL_P1.f19_g16_rx1.A.yellowstone_intel in 1.374002 seconds (PASS)
Finished XML for test TESTRUNPASS_P1.f19_g16_rx1.A.yellowstone_intel in 1.421210 seconds (PASS)
Finished XML for test TESTRUNFAIL_P1.f19_g16_rx1.A.yellowstone_intel in 1.425648 seconds (PASS)
Starting SETUP for test TESTBUILDFAIL_P1.f19_g16_rx1.A.yellowstone_intel with 1 procs
Starting SETUP for test TESTRUNFAIL_P1.f19_g16_rx1.A.yellowstone_intel with 1 procs
Starting SETUP for test TESTRUNPASS_P1.f19_g16_rx1.A.yellowstone_intel with 1 procs
Finished SETUP for test TESTBUILDFAIL_P1.f19_g16_rx1.A.yellowstone_intel in 3.956019 seconds (PASS)
Starting SHAREDLIB_BUILD for test TESTBUILDFAIL_P1.f19_g16_rx1.A.yellowstone_intel with 1 procs
Finished SETUP for test TESTRUNPASS_P1.f19_g16_rx1.A.yellowstone_intel in 4.023581 seconds (PASS)
Finished SETUP for test TESTRUNFAIL_P1.f19_g16_rx1.A.yellowstone_intel in 4.076013 seconds (PASS)
Finished SHAREDLIB_BUILD for test TESTBUILDFAIL_P1.f19_g16_rx1.A.yellowstone_intel in 1.534378 seconds (PASS)
Starting MODEL_BUILD for test TESTBUILDFAIL_P1.f19_g16_rx1.A.yellowstone_intel with 4 procs
Starting SHAREDLIB_BUILD for test TESTRUNFAIL_P1.f19_g16_rx1.A.yellowstone_intel with 1 procs
Finished MODEL_BUILD for test TESTBUILDFAIL_P1.f19_g16_rx1.A.yellowstone_intel in 1.439846 seconds (FAIL). [COMPLETED 1 of 3]
    Case dir: /glade/scratch/jedwards/scripts_regression_test.20171109_140320/TESTBUILDFAIL_P1.f19_g16_rx1.A.yellowstone_intel.20171109_140326
    Errors were:
        Building test for TESTBUILDFAIL in directory /glade2/scratch2/jedwards/scripts_regression_test.20171109_140320/TESTBUILDFAIL_P1.f19_g16_rx1.A.yellowstone_intel.20171109_140326
        ERROR: BUILD FAIL: Intentional fail for testing infrastructure

Finished SHAREDLIB_BUILD for test TESTRUNFAIL_P1.f19_g16_rx1.A.yellowstone_intel in 1.488618 seconds (PASS)
Starting MODEL_BUILD for test TESTRUNFAIL_P1.f19_g16_rx1.A.yellowstone_intel with 4 procs
Starting SHAREDLIB_BUILD for test TESTRUNPASS_P1.f19_g16_rx1.A.yellowstone_intel with 1 procs
Finished MODEL_BUILD for test TESTRUNFAIL_P1.f19_g16_rx1.A.yellowstone_intel in 1.419215 seconds (PASS)
Finished SHAREDLIB_BUILD for test TESTRUNPASS_P1.f19_g16_rx1.A.yellowstone_intel in 1.518366 seconds (PASS)
Starting MODEL_BUILD for test TESTRUNPASS_P1.f19_g16_rx1.A.yellowstone_intel with 4 procs
Starting RUN for test TESTRUNFAIL_P1.f19_g16_rx1.A.yellowstone_intel with 1 proc on interactive node and 1 procs on compute nodes
Finished MODEL_BUILD for test TESTRUNPASS_P1.f19_g16_rx1.A.yellowstone_intel in 1.567670 seconds (PASS)
Starting RUN for test TESTRUNPASS_P1.f19_g16_rx1.A.yellowstone_intel with 1 proc on interactive node and 1 procs on compute nodes
Finished RUN for test TESTRUNFAIL_P1.f19_g16_rx1.A.yellowstone_intel in 7.315589 seconds (PEND). [COMPLETED 2 of 3]
Finished RUN for test TESTRUNPASS_P1.f19_g16_rx1.A.yellowstone_intel in 6.645838 seconds (PEND). [COMPLETED 3 of 3]
Waiting for tests to finish
Test 'TESTBUILDFAIL_P1.f19_g16_rx1.A.yellowstone_intel' finished with status 'FAIL'
    Path: /glade/scratch/jedwards/scripts_regression_test.20171109_140320/TESTBUILDFAIL_P1.f19_g16_rx1.A.yellowstone_intel.20171109_140326/TestStatus
Test 'TESTRUNFAIL_P1.f19_g16_rx1.A.yellowstone_intel' finished with status 'FAIL'
    Path: /glade/scratch/jedwards/scripts_regression_test.20171109_140320/TESTRUNFAIL_P1.f19_g16_rx1.A.yellowstone_intel.20171109_140326/TestStatus
Test 'TESTRUNPASS_P1.f19_g16_rx1.A.yellowstone_intel' finished with status 'PASS'
    Path: /glade/scratch/jedwards/scripts_regression_test.20171109_140320/TESTRUNPASS_P1.f19_g16_rx1.A.yellowstone_intel.20171109_140326/TestStatus
Using project from env PROJECT: CESM0008
Using existing test directory /glade/scratch/jedwards/scripts_regression_test.20171109_140320/TESTBUILDFAIL_P1.f19_g16_rx1.A.yellowstone_intel.20171109_140326
Using existing test directory /glade/scratch/jedwards/scripts_regression_test.20171109_140320/TESTRUNFAIL_P1.f19_g16_rx1.A.yellowstone_intel.20171109_140326
Test TESTRUNPASS_P1.f19_g16_rx1.A.yellowstone_intel passed and will not be re-run
Using existing test directory /glade/scratch/jedwards/scripts_regression_test.20171109_140320/TESTRUNPASS_P1.f19_g16_rx1.A.yellowstone_intel.20171109_140326
RUNNING TESTS:
  TESTBUILDFAIL_P1.f19_g16_rx1.A.yellowstone_intel
  TESTRUNFAIL_P1.f19_g16_rx1.A.yellowstone_intel
  TESTRUNPASS_P1.f19_g16_rx1.A.yellowstone_intel
Starting MODEL_BUILD for test TESTBUILDFAIL_P1.f19_g16_rx1.A.yellowstone_intel with 4 procs
Starting RUN for test TESTRUNFAIL_P1.f19_g16_rx1.A.yellowstone_intel with 1 procs
Finished MODEL_BUILD for test TESTBUILDFAIL_P1.f19_g16_rx1.A.yellowstone_intel in 8.656065 seconds (PASS)
Finished RUN for test TESTRUNFAIL_P1.f19_g16_rx1.A.yellowstone_intel in 8.660039 seconds (PEND). [COMPLETED 1 of 3]
Starting RUN for test TESTBUILDFAIL_P1.f19_g16_rx1.A.yellowstone_intel with 1 proc on interactive node and 1 procs on compute nodes
Finished RUN for test TESTBUILDFAIL_P1.f19_g16_rx1.A.yellowstone_intel in 6.375755 seconds (PEND). [COMPLETED 2 of 3]
Waiting for tests to finish
Test 'TESTBUILDFAIL_P1.f19_g16_rx1.A.yellowstone_intel' finished with status 'PASS'
    Path: /glade/scratch/jedwards/scripts_regression_test.20171109_140320/TESTBUILDFAIL_P1.f19_g16_rx1.A.yellowstone_intel.20171109_140326/TestStatus
Test 'TESTRUNFAIL_P1.f19_g16_rx1.A.yellowstone_intel' finished with status 'FAIL'
    Path: /glade/scratch/jedwards/scripts_regression_test.20171109_140320/TESTRUNFAIL_P1.f19_g16_rx1.A.yellowstone_intel.20171109_140326/TestStatus
Test 'TESTRUNPASS_P1.f19_g16_rx1.A.yellowstone_intel' finished with status 'PASS'
    Path: /glade/scratch/jedwards/scripts_regression_test.20171109_140320/TESTRUNPASS_P1.f19_g16_rx1.A.yellowstone_intel.20171109_140326/TestStatus
ERRPUT: 
```

--- a/scripts/lib/CIME/test_scheduler.py
+++ b/scripts/lib/CIME/test_scheduler.py
@@ -221,21 +221,26 @@ class TestScheduler(object):
 
         if use_existing:
             for test in self._tests:
-                ts = TestStatus(self._get_test_dir(test))
-                for phase, status in ts:
-                    if phase in CORE_PHASES:
-                        if status in [TEST_PEND_STATUS, TEST_FAIL_STATUS]:
-                            # We need to pick up here
-                            break
-                        else:
-                            if phase != SUBMIT_PHASE:
-                                # Somewhat subtle. Create_test considers submit/run to be the run phase,
-                                # so don't try to update test status for a passed submit phase
-                                self._update_test_status(test, phase, TEST_PEND_STATUS)
-                                self._update_test_status(test, phase, status)
+                with TestStatus(self._get_test_dir(test)) as ts:
+                    for phase, status in ts:
+                        if phase in CORE_PHASES:
+                            if status in [TEST_PEND_STATUS, TEST_FAIL_STATUS]:
+                                if status == TEST_FAIL_STATUS:
+                                    # Import for potential subsequent waits
+                                    ts.set_status(phase, TEST_PEND_STATUS)
 
-                                if phase == RUN_PHASE:
-                                    logger.info("Test {} passed and will not be re-run".format(test))
+                                # We need to pick up here
+                                break
+
+                            else:
+                                if phase != SUBMIT_PHASE:
+                                    # Somewhat subtle. Create_test considers submit/run to be the run phase,
+                                    # so don't try to update test status for a passed submit phase
+                                    self._update_test_status(test, phase, TEST_PEND_STATUS)
+                                    self._update_test_status(test, phase, status)
+
+                                    if phase == RUN_PHASE:
+                                        logger.info("Test {} passed and will not be re-run".format(test))
 
                 logger.info("Using existing test directory {}".format(self._get_test_dir(test)))
         else:

